src/harness/BenchmarkDotNet.Extensions/ValuesGenerator.cs

```diff
@@ -21,6 +22,22 @@ public static T GetNonDefaultValue<T>()
                 return ArrayOfUniqueValues<T>(2).First(value => !value.Equals(default));
         }
 
+        /// <summary>
+        /// generates an array of unique random values
+        /// the array is guaranteed to be not 16-byte aligned
+        /// </summary>
+        /// <remarks>make sure to Free the handle in a Cleanup method</remarks>
+        public static T[] UnalignedArrayOfUniqueValues<T>(int count, out GCHandle pinnedArrayHandle)
```

Comment:
As indicated in the comment on the main thread, I do not think this works as intended.